DOC.md

```diff
@@ -1233,17 +1233,70 @@ ls.parser.parse_snippet({trig = "lsp"}, "$1 is ${2|hard,easy,challenging|}")
 
 <!-- panvimdoc-ignore-end -->
 
+`context` can be:
+  - `string|table`: treated like the first argument to `ls.s`, `parse_snippet`
+    returns a snippet.
+  - `number`: `parse_snippet` returns a snippetNode, with the position
+    `context`.
+  - `nil`: `parse_snippet` returns a flat table of nodes. This can be used
+    like `fmt`.
+
 Nested placeholders(`"${1:this is ${2:nested}}"`) will be turned into
 choiceNode's with:
-	- the given snippet(`"this is ${1:nested}"`) and
-	- an empty insertNode
+  - the given snippet(`"this is ${1:nested}"`) and
+  - an empty insertNode
 
 <!-- panvimdoc-ignore-start -->
 
 ![lsp2](https://user-images.githubusercontent.com/25300418/184359306-c669d3fa-7ae5-4c07-b11a-34ae8c4a17ac.gif)
 
 <!-- panvimdoc-ignore-end -->
 
+This behaviour can be modified by changing `parser_nested_assembler` in
+`ls.setup()`.
+
+
+Luasnip will also modify some snippets it's incapable of representing
+accurately:
+  - if the `$0` is a placeholder with something other than just text inside
+  - if the `$0` is a choice
+  - if the `$0` is not an immediate child of the snippet (it could be inside a
+    placeholder: `"${1: $0 }"`)
+
+To remedy those incompatibilities, the invalid `$0` will be replaced with a
+tabstop/placeholder/choice which will be visited just before the new `$0`. This
+new `$0` will be inserted at the (textually) earliest valid position behind the
+invalid `$0`.
+
+## Snipmate
+
+It is furthermore possible to parse snipmate-snippets (this includes support for
+vimscript-evaluation!!)  
+Snipmate-snippets have to be parsed with a different function,
+`ls.parser.parse_snipmate`:
+```lua
+ls.parser.parse_snipmate("year", "The year is `strftime('%Y')`")
+```
+
+`parse_snipmate` is the same as `parse_snippet`, only the snippet-body is parsed
+differently.
+
+## Transformations
+
+To apply
+[Variable/Placeholder-transformations](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_variable-transforms),
+luasnip needs to apply ECMAScrip-regexes.  
+This is implemented by relying on [`jsregexp`](https://github.com/kmarius/jsregexp).  
+`jsregexp` is available as a luarock, and, as such, can be installed by
+`packer.nvim` (although actual usage may require a small workaround, see
+[here](https://github.com/wbthomason/packer.nvim/issues/593) or
+[here](https://github.com/wbthomason/packer.nvim/issues/358)).  
+Alternatively, the repo can be cloned, `make`d, and the resulting `jsregexp.so`
+placed in some place where nvim can find it (probably `~/.config/nvim/lua/`).
+
+If `jsregexp` cannot be found, luasnip and omits the transformation and just
```

Comment:
incomplete sentence? (after `luasnip`)

Reply:
Ah, I reworded it a few times, looks like I missed that :D
TY!